### PR TITLE
Fix preserving timestamp when the move command is executing

### PR DESCRIPTION
**Pull request by KostiantynPopovych**
This PR resolves issue: #992.

These test files are pretty similar, I just took the `copy` method test files as a blueprint, and it seems we can make some refactoring here and there. I'm open to any suggestions!

### CODE REVIEW

**Comment by RyanZim** [comment]
https://github.com/jprichardson/node-fs-extra/issues/992#issuecomment-1414186359

**Comment by KostiantynPopovych** [comment]
Hey @RyanZim, I updated the implementation to match the logic that was discussed in thread #992. I refactored a bit methods move & moveSync to match the same style.

**Comment by RyanZim** [comment]
It's not clear to me why the large refactor is needed; theoretically you only need to set `preserveTimestamps` in `moveAcrossDevice`, similarly to how `errorOnExist` is currently set. What am I missing?

**Comment by KostiantynPopovych** [comment]
@RyanZim Reverted original code, that refactoring might be overengineering for now 🤔

File: lib/move/__tests__/move-preserve-timestamp.test.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+const fs = require('../../')
+const os = require('os')
+const path = require('path')
+const utimesSync = require('../../util/utimes').utimesMillisSync
+const assert = require('assert')
+const fse = require('../../index')
+
+/* global beforeEach, afterEach, describe, it */
+
+if (process.arch === 'ia32') console.warn('32 bit arch; skipping copy timestamp tests')
```
Comment (RyanZim) [inline lib/move/__tests__/move-preserve-timestamp.test.js:12]:
```suggestion
if (process.arch === 'ia32') console.warn('32 bit arch; skipping move timestamp tests')
```

File: lib/move/__tests__/move-preserve-timestamp.test.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      }
+    })
+  }
+
+  const _it = __skipTests ? it.skip : it
+
+  afterEach(done => {
+    fse.removeSync(TEST_DIR)
+    fse.removeSync(SRC)
+    done()
```
Comment (RyanZim) [inline lib/move/__tests__/move-preserve-timestamp.test.js:58]: Don't need `done` for synchronous hooks
```suggestion
  afterEach(() => {
    fse.removeSync(TEST_DIR)
    fse.removeSync(SRC)
```

File: lib/move/__tests__/move-sync-preserve-timestamp.test.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+const fs = require('../../')
+const os = require('os')
+const path = require('path')
+const utimesSync = require('../../util/utimes').utimesMillisSync
+const assert = require('assert')
+const fse = require('../../index')
+
+/* global beforeEach, afterEach, describe, it */
+
+if (process.arch === 'ia32') console.warn('32 bit arch; skipping copy timestamp tests')
```
Comment (RyanZim) [inline lib/move/__tests__/move-sync-preserve-timestamp.test.js:12]:
```suggestion
if (process.arch === 'ia32') console.warn('32 bit arch; skipping move timestamp tests')
```

File: lib/move/__tests__/move-sync-preserve-timestamp.test.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+      }
+    })
+  }
+
+  const _it = __skipTests ? it.skip : it
+
+  afterEach(done => {
+    fse.removeSync(TEST_DIR)
+    fse.removeSync(SRC)
+    done()
```
Comment (RyanZim) [inline lib/move/__tests__/move-sync-preserve-timestamp.test.js:58]:
```suggestion
  afterEach(() => {
    fse.removeSync(TEST_DIR)
    fse.removeSync(SRC)
```

**Review by RyanZim — COMMENTED** [review]
A few nits in the tests; otherwise looks good.

**Comment by KostiantynPopovych** [comment]
Thank you for reviewing it!

**Pull request merged**